api.ts

```diff
@@ -2249,6 +2292,36 @@ export function makeSyncTable<
       }
     : undefined;
 
+  const executeGetPermissions = wrappedExecuteGetPermissions
+    ? async function execGetPermissions(
+        rows: Array<ObjectSchemaDefinitionType<K, L, SchemaDefT>>,
+        context: GetPermissionExecutionContext,
+      ) {
+        const result = await wrappedExecuteGetPermissions(rows, context);
+        const {permissions} = result;
+        const permissionCountByRow: {[rowId: string]: number} = permissions.reduce(
+          (acc: {[rowId: string]: number}, permission: Permission) => {
+            acc[permission.rowId] = acc[permission.rowId]++ || 1;
+            return acc;
+          },
+          {},
+        );
+
+        const oversizedRowIds = Object.entries(permissionCountByRow)
+          .filter(([_rowId, count]) => count > MaxPermissionsPerRow)
+          .map(([rowId, _count]) => rowId);
+
+        if (oversizedRowIds.length > 0) {
+          throw new Error(
+            `Permissions limit exceeded limit of ${MaxPermissionsPerRow} permissions for row with ids: ${oversizedRowIds.join(
+              ', ',
+            )}`,
+          );
+        }
+        return result;
```

Comment:
We should likely log the full permissions when this error occurs. Would be good to do this the canonical way

Reply:
What if there are 1000 rows that are oversized?  Do we want to have all this in the error?  Maybe we should add a property onto the error, but only show the first 10 or some number?

Reply:
Yeah good question. We could set some reasonable limit on what we log here. Will wait on some input from others but fine doing a limit